integration_tests/src/main/python/parquet_test.py

```diff
@@ -728,3 +728,39 @@ def do_parquet_scan(spark):
         exist_classes= "BatchScanExec",
         non_exist_classes= "GpuBatchScanExec",
         conf = conf_for_parquet_aggregate_pushdown)
+
+@pytest.mark.skipif(is_before_spark_330(), reason='Hidden file metadata columns are a new feature of Spark 330')
+@allow_non_gpu(any = True)
+def test_parquet_scan_with_hidden_metadata_fallback(spark_tmp_path):
+    data_path = spark_tmp_path + "/hidden_metadata.parquet"
+    with_cpu_session(lambda spark : spark.range(10) \
+                     .selectExpr("id", "id % 3 as p") \
+                     .write \
+                     .partitionBy("p") \
+                     .mode("overwrite") \
+                     .parquet(data_path))
+
+    def do_parquet_scan(spark):
+        df = spark.read.parquet(data_path).selectExpr("id", "_metadata.file_path", "_metadata.file_name",
+                                                      "_metadata.file_size", "_metadata.file_modification_time")
+        return df
+
+    assert_cpu_and_gpu_are_equal_collect_with_capture(
+        do_parquet_scan,
+        exist_classes= "FileSourceScanExec",
+        non_exist_classes= "GpuBatchScanExec")
+
+@pytest.mark.skipif(is_before_spark_330(), reason='Hidden file metadata columns are a new feature of Spark 330')
+def test_parquet_scan_with_hidden_metadata_column_names(spark_tmp_path):
+    """
+    If the parquet file has column names file_path, file_name, file_size, file_modification_time then these
+    override the hidden metadata columns and we can run on GPU
+    """
+    data_path = spark_tmp_path + "/hidden_metadata_column_names.parquet"
+
+    def do_parquet_scan(spark):
+        spark.range(10).selectExpr("id", "id % 3 as file_size").write.partitionBy("id").mode("overwrite").parquet(data_path)
+        df = spark.read.parquet(data_path).selectExpr("Max(file_size)")
```

Comment:
I'm confused by this test, as it's not trying to access a metadata column via the `_metadata.` prefix so I'm not sure how it would potentially trigger the metadata path.  I'd actually be more curious what would happen if the Parquet file had a column named `_metadata`, especially if it is a struct column with fields named `file_size`, etc.